wordpress-mobile/WordPress-Android
# Feature Branch - Implementation more menu for Dashboard Cards

## Description
This PR  
1. Adds the more menu to all the dashboard cards 

| Card  | More menu |
|---|---|
| Activity Log Card | <img src= "https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/db3e97ad-50ae-4600-a4ac-e473c5a127e9" width="220" height="480">|
| Pages Card| <img src ="https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/5ce7fff7-6b52-4c3b-8c11-46d916a3527d" width="220" height="480">|
| Today Stats Card | <img src="https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/b1936e28-73b5-4554-a667-1995d0266b3b"  width="220" height="480">|   
| Blaze Card(Promote) | <img src = "https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/37954829-ca03-4c07-a34e-e0ceeb224b17"  width="220" height="480">|
| Blaze Card(Campaign) | <img src = "https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/a4521b9e-f6b3-4fa5-9e22-dc22a6f8f2cb"  width="220" height="480">| 
| Draft Posts Card | <img src = "https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/2fa44ece-e289-43fb-80b6-612c6f52e7cf"  width="220" height="480">|
| Scheduled Posts Card | <img src="https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/2806be6d-52dc-4e2c-90c2-dc13ef517001"  width="220" height="480"> |
| Quick Start Card|  <img src="https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/300dd1cd-f708-48e5-a1ab-5ebb96c89eef" width="220" height="480"> |

3. Adds **hide this** option to all the dashboard cards 
4. Adds more menu options to all the dashboard cards 
5. Removes footer from Today's stats, Activity log, Draft Post Card and Scheduled Post card

## Not included in the PR 
The feature to unhide the cards once they are hidden is in progress. 

## To test:
1. Login to the Jetpack app 
3. Go to MySiteDashboard 
4. Verify the functionality of each card 
5. Click on the more menu of each card and verify that it is working as expected
7. Click on "hide this" for each card and verify that the card is hidden for the site 

## This PR includes the following PR's 
- https://github.com/wordpress-mobile/WordPress-Android/pull/18968
- https://github.com/wordpress-mobile/WordPress-Android/pull/19005
- https://github.com/wordpress-mobile/WordPress-Android/pull/19008
- https://github.com/wordpress-mobile/WordPress-Android/pull/19112
- https://github.com/wordpress-mobile/WordPress-Android/pull/19011
- https://github.com/wordpress-mobile/WordPress-Android/pull/18994
- https://github.com/wordpress-mobile/WordPress-Android/pull/19020
- https://github.com/wordpress-mobile/WordPress-Android/pull/19055
- https://github.com/wordpress-mobile/WordPress-Android/pull/19042
- https://github.com/wordpress-mobile/WordPress-Android/pull/19043
- https://github.com/wordpress-mobile/WordPress-Android/pull/19047
- https://github.com/wordpress-mobile/WordPress-Android/pull/19053
- https://github.com/wordpress-mobile/WordPress-Android/pull/19058
- https://github.com/wordpress-mobile/WordPress-Android/pull/19059
- https://github.com/wordpress-mobile/WordPress-Android/pull/19056
- https://github.com/wordpress-mobile/WordPress-Android/pull/19063
- https://github.com/wordpress-mobile/WordPress-Android/pull/19099
- https://github.com/wordpress-mobile/WordPress-Android/pull/19075
- https://github.com/wordpress-mobile/WordPress-Android/pull/19089
- https://github.com/wordpress-mobile/WordPress-Android/pull/19090
- https://github.com/wordpress-mobile/WordPress-Android/pull/19091
- https://github.com/wordpress-mobile/WordPress-Android/pull/19104
- https://github.com/wordpress-mobile/WordPress-Android/pull/19107


## ⚠️  Merge Instructions 
- Merge FLUXC PR https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2814
- Update the FluxC Hash version 
- Remove Donot Merge label
- Wait for CI to Complete
- Merge as normal

Note: Since this PR implements the Hide functionality only, and Unhide functionality is in Progress. Please wait untill the release branch is cut for 23.2 to merge this PR

## Regression Notes
1. Potential unintended areas of impact
- Dashboard cards are not working as expected

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit tests and Manual tests

3. What automated tests I added (or what prevented me from doing so)
Unit tests

PR submission checklist:

UI Changes testing checklist:
